plugins/uefi/fu-uefi-common.c

```diff
@@ -241,12 +244,17 @@ fu_uefi_get_esp_path_for_os (const gchar *base)
 		return g_steal_pointer (&esp_path);
 	/* if ID key doesn't exist, try ID_LIKE */
 	id_like_id = g_hash_table_lookup (os_release, "ID_LIKE");
+	/* try to fallback to use UEFI removable path if ID_LIKE key doesn't exist */
 	if (id_like_id != NULL) {
 		g_autofree gchar* id_like_path = g_build_filename (base, "EFI", id_like_id, NULL);
 		if (g_file_test (id_like_path, G_FILE_TEST_IS_DIR)) {
 			g_debug ("Using ID_LIKE key from os-release");
 			return g_steal_pointer (&id_like_path);
 		}
+	} else if (fu_device_get_metadata_boolean (device, "FallbacktoRemovablePath")) {
```

Comment:
I think this specific clause needs to just be an `if` not a `else if`.  That way even if `ID_LIKE` key was set but it didn't exist, we fallback.  Right now if you have `ID_LIKE` key but file doesn't and you set option exist you would return `unknown`.
I think it should be:
1. Try `ID`
2. Try `ID_LIKE`
3. Check for removable path option
4. If option set, try removable path
5. Return unknown if all fail

Reply:
Thanks for pointing out this pitfall. Fixed.